defos/src/defos_linux.cpp

```diff
@@ -512,6 +515,22 @@ void defos_update()
 
     window_has_focus = focused_window == win;
     apply_cursor_visible();
+
+	// Cursor enter/exit events
+	bool state = defos_is_mouse_in_view();
+	if (state != is_cursor_in_view && window_has_focus)
+	{
+		switch(state)
```

Comment:
Don't use a `switch` for a boolean value. Use and `if` statement or a ternary

Reply:
One sec. I'll rewrite this part for you.

Reply:
Done

Reply:
Can you please check if it works for you? I don't have access to Linux right now.

Reply:
Tried to test last night but ran into complications and couldn't get to it.
Thank you for doing that.
Almost perfect, but flip the events and put the variable swap afterwards.
```lua

bool visible = defos_is_mouse_in_view() && window_has_focus;
if (visible != is_cursor_in_view)
{
    defos_emit_event(is_cursor_in_view ? DEFOS_EVENT_MOUSE_LEAVE : DEFOS_EVENT_MOUSE_ENTER);
    is_cursor_in_view = visible;
}
```

Reply:
Why? It's logically the same thing and making the assignment early means weird things cannot happen in `defos_emit_event`, which calls into the Lua side (though in this particular instance, I don't see any way it could go wrong).

Reply:
Oh. Oops. I wrote `current_state` instead of `visible`

Reply:
Ah, you're right. When I tested I thought the assignment needed to go after so it seemed to reverse it once I had moved it, but with the assignment before it works fine.

So yes, that looks fine. Ill get to the other logic soon.